source/crc32.cpp

```diff
@@ -57,36 +57,39 @@ static const unsigned long crcTable[256] = {
     0x2d02ef8dUL
 };
 
-unsigned long crc32FromUpcaseString(const char *str)
-{
-    unsigned long crc = 0xFFFFFFFF;
-    while (*str)
-        crc = crcTable[(crc^toupper(*str++)) & 0xff] ^ (crc >> 8);
-    return crc;
-}
 
-unsigned long crc32FromUpcaseStdString(const std::string& str)
-{
-    return crc32FromUpcaseString(str.c_str());
-}
+unsigned long crc32FromUpcaseString(const char *str) {
+        unsigned long crc = 0xFFFFFFFF;
+        while (*str) {
+            crc = crcTable[(crc^toupper(*str++)) & 0xff] ^ (crc >> 8);
+        }
+        return crc;
+    }
 
-unsigned long crc32FromString(const char *str)
-{
-    unsigned long crc = 0xFFFFFFFF;
-    while (*str)
-        crc = crcTable[(crc ^ (*str++)) & 0xff] ^ (crc >> 8);
-    return crc;
-}
+unsigned long crc32FromUpcaseStdString(const std::string& str) {
+        return crc32FromUpcaseString(str.c_str());
+    }
 
-unsigned long crc32FromStdString(const std::string& str)
-{
-    return crc32FromString(str.c_str());
-}
+unsigned long crc32FromString(const char *str) {
+        unsigned long crc = 0xFFFFFFFF;
+        while (*str) {
+            crc = crcTable[(crc ^ (*str++)) & 0xff] ^ (crc >> 8);
+        }
+        return crc;
+    }
+
+unsigned long crc32FromStdString(const std::string& str) {
+        return crc32FromString(str.c_str());
+    }
 
-unsigned long crc32(const unsigned char *buf, unsigned long len)
-{
-    unsigned long crc = 0xFFFFFFFF;
-    for (unsigned i = 0; i < len; i++)
-        crc = crcTable[(crc^buf[i]) & 0xFF] ^ (crc >> 8);
-    return crc;
+unsigned long crc32(const unsigned char *buf, unsigned long len) {
+        if (len > sizeof(crcTable)) {
```

Comment:
Hmm, that's interesting though. At first I hoped that crc32 is no longer used in CLEO core, but:
https://github.com/cleolibrary/CLEO5/blob/master/source/CScriptEngine.cpp#L1552
where majority of scripts is longer than 256 bytes, so it uses random data as CRC32 table.

This checksum is used by CScriptEngine::SaveState() which opens whole can of worms. This need further investigation. Is saving loading state even working right now? It will depend on fact how static is the data in memory after CRC table.

Reply:
Oh, ok the table is addressed with modulo of 255:
`crcTable[x & 0xFF]` so no memory overrun happens.